cpp/include/cuspatial/detail/utility/validation.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuspatial/error.hpp>
+
+/**
+ * @brief Macro for validating the data array sizes for a polygon.
+ *
+ * Raises an exception if any of the following are false:
+ *  - There are either zero or at least 2 polygon offsets (number of polygons + 1).
+ *  - There are either zero or at least 2 polygon ring offsets (number of rings + 1).
```

Comment:
As discussed in https://github.com/rapidsai/cuspatial/discussions/972, there should be at least 1 element in offset array.